airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/avro_parser.py

```diff
@@ -138,15 +139,20 @@ def parse_records(
         if not isinstance(avro_format, AvroFormat):
             raise ValueError(f"Expected ParquetFormat, got {avro_format}")
 
-        with stream_reader.open_file(file, self.file_read_mode, self.ENCODING, logger) as fp:
-            avro_reader = fastavro.reader(fp)
-            schema = avro_reader.writer_schema
-            schema_field_name_to_type = {field["name"]: field["type"] for field in schema["fields"]}
-            for record in avro_reader:
-                yield {
-                    record_field: self._to_output_value(avro_format, schema_field_name_to_type[record_field], record[record_field])
-                    for record_field, record_value in schema_field_name_to_type.items()
-                }
+        line_no = 0
+        try:
+            with stream_reader.open_file(file, self.file_read_mode, self.ENCODING, logger) as fp:
+                avro_reader = fastavro.reader(fp)
+                schema = avro_reader.writer_schema
+                schema_field_name_to_type = {field["name"]: field["type"] for field in schema["fields"]}
+                for record in avro_reader:
+                    line_no += 1
+                    yield {
+                        record_field: self._to_output_value(avro_format, schema_field_name_to_type[record_field], record[record_field])
+                        for record_field, record_value in schema_field_name_to_type.items()
+                    }
+        except Exception:
+            raise RecordParseError(FileBasedSourceError.ERROR_PARSING_RECORD, filename=file.uri, lineno=line_no)
```

Comment:
can't we raise this error from the stream class to avoid implementing this in all parsers?

Reply:
It's tricky to implement when the actual parser doesn't raise the exception, could you please advise where would be the perfect place from your point of view to have the exception raised? @girarda

Reply:
@bazarnov would it make sense to put it [here](https://github.com/airbytehq/airbyte/blob/d97a399a24e5d86d4a0fcedb2b8055f7ede834eb/airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py#L115)?

Reply:
@clnoll I've corrected the flow and "yes", we are collecting errors from the place you've mentioned.